include/envoy/compressor/compressor.h

```diff
@@ -0,0 +1,39 @@
+#pragma once
+
+#include "envoy/buffer/buffer.h"
+
+namespace Envoy {
+namespace Compressor {
+
+/**
+ * Public compressor allows compress and decompress buffer data.
+ */
+class Compressor {
+public:
+  virtual ~Compressor() {}
+
+  /**
+   * Compress data from one buffer into another buffer
+   * @param in supplies the buffer with data to be compressed
+   * @param out supplies the buffer to output compressed data
+   * @return bool indicating whether operation succeeded or not.
+   */
+  virtual bool compress(Buffer::Instance& in, Buffer::Instance& out) PURE;
```

Comment:
After reading the test, it looks like the input Buffer will actually be modified after the call so const wouldn't be appropriate. This wasn't clear from the comment. If this is the desired behavior can you expand the comment to clarify? From the test it looks like after the call the input buffer should always be empty.